DirectProgramming/DPC++FPGA/Tutorials/Features/private_copies/src/private_copies.cpp

```diff
@@ -85,95 +85,91 @@ void PartialSumWithShift(const device_selector &selector, const IntArray &array,
-  std::cout << "Max concurrency " << concurrency << " "
+  std::cout << "Num private_copies " << num_copies << " "
             << "kernel time : " << kernel_time << " ms\n";
-  std::cout << "Throughput for kernel with max_concurrency " << concurrency
+  std::cout << "Throughput for kernel with private_copies " << num_copies
             << ": ";
   std::cout << std::fixed << std::setprecision(3)
-            << ((double)(kTotalOps) / kernel_time) / 1e3f << " MIPS\n";
+            << ((double)(kTotalOps) / kernel_time) / 1e6f << " GFlops\n";
 }
 
 // Calculates the expected results. Used to verify that the kernel
 // is functionally correct.
-int GoldenResult(const IntArray &A, int shift) {
+int GoldenResult(const IntArray &input_arr, int shift) {
   int gr = 0;
+
   for (size_t i = 0; i < kMaxIter; i++) {
-    int a1[kSize];
-    for (size_t j = 0; j < kSize; j++) a1[j] = A[(i * 4 + j) % kSize] * shift;
-    for (size_t j = 0; j < kSize; j++) gr += a1[j];
+    int a[kSize];
+    for (size_t j = 0; j < kSize; j++) {
+      a[j] = input_arr[(i * 4 + j) % kSize] * shift;
+    }
+    for (size_t j = 0; j < kSize; j++) 
+      gr += a[j];
   }
+
   return gr;
 }
 
 int main() {
   bool success = true;
 
-  IntArray A;
-  IntScalar R0, R1, R2, R3, R4, R5;
+  IntArray a;
+  IntScalar R0, R1, R2, R3, R4;
 
   int shift = rand() % kMaxValue;
 
   // initialize the input data
-  for (size_t i = 0; i < kSize; i++) A[i] = rand() % kMaxValue;
+  for (size_t i = 0; i < kSize; i++)
+    a[i] = rand() % kMaxValue;
 
 #if defined(FPGA_EMULATOR)
   INTEL::fpga_emulator_selector selector;
 #else
   INTEL::fpga_selector selector;
 #endif
 
-  // Run the kernel with different values of the max_concurrency
-  // attribute, to determine the optimal concurrency.
-  // In this case, the optimal max_concurrency is 2 since this
-  // achieves the highest throughput. Higher values of max_concurrency
-  // consume additional RAM without increasing throughput.
-  PartialSumWithShift<0>(selector, A, shift, R0);
-  PartialSumWithShift<1>(selector, A, shift, R1);
-  PartialSumWithShift<2>(selector, A, shift, R2);
-  PartialSumWithShift<4>(selector, A, shift, R3);
-  PartialSumWithShift<8>(selector, A, shift, R4);
-  PartialSumWithShift<16>(selector, A, shift, R5);
+  // Run the kernel with different values of the private_copies
+  // attribute, to determine the optimal private_copies number.
+  SimpleMathWithShift<0>(selector, a, shift, R0);
+  SimpleMathWithShift<1>(selector, a, shift, R1);
+  SimpleMathWithShift<2>(selector, a, shift, R2);
+  SimpleMathWithShift<3>(selector, a, shift, R3);
+  SimpleMathWithShift<4>(selector, a, shift, R4);
 
   // compute the actual result here
-  int gr = GoldenResult(A, shift);
+  int gr = GoldenResult(a, shift);
 
   // verify the results are correct
   if (gr != R0[0]) {
-    std::cout << "Max Concurrency 0: mismatch: " << R0[0] << " != " << gr
+    std::cout << "private_copies 0: mismatch: " << R0[0] << " != " << gr
```

Comment:
```suggestion
    std::cout << "Private copies 0: mismatch: " << R0[0] << " != " << gr
```
Same for the others.